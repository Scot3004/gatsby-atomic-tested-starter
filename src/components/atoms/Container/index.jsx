import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

const MainStyle = styled.div`
  display: grid;
  grid-template-columns: minmax(auto, 1024px);
  justify-content: center;
  padding: 1em;
`

const MainContainer = styled.div`
  display: grid;
  grid-template-columns: 1fr;
`

const Main = ({ children }) => (
  <MainStyle>
    <MainContainer>
      {children}
    </MainContainer>
  </MainStyle>
)

export default Main

Main.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
}
