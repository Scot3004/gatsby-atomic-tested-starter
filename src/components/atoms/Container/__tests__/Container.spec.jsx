import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import Container from '..';

test('it works', () => {
  const tree = renderer.create(<Container>container stub</Container>).toJSON();
  expect(tree).toMatchSnapshot();
});
