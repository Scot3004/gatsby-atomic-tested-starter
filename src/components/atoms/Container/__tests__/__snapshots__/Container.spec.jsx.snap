// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
.c0 {
  display: grid;
  grid-template-columns: minmax(auto,64em);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem;
}

.c1 {
  display: grid;
  grid-template-columns: 1fr;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    container stub
  </div>
</div>
`;
